garak/generators/langchain_serve.py

```diff
@@ -28,20 +28,27 @@ class LangChainServeLLMGenerator(Generator):
     """
 
     generator_family_name = "LangChainServe"
+    DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {"config_hash": "default"}
+
     config_hash = "default"
 
     def __init__(
-        self, name=None, generations=10
+        self, name=None, generations=10, config_root=_config
     ):  # name not required, will be extracted from uri
+        self.uri = None
+        self._load_config(config_root)
         self.generations = generations
-        api_uri = os.getenv("LANGCHAIN_SERVE_URI")
-        if not self._validate_uri(api_uri):
+        if self.uri is None:
+            self.uri = os.getenv("LANGCHAIN_SERVE_URI")
+        if not self._validate_uri(self.uri):
```

Comment:
I see what you are referring to, at the class level `uri` is being reserved in some plugins. 

I land the other direction, I would say the `uri` attribute should not be reserved for "references" and it would be more clear to define specific documentation in an attribute with a lower probability for collision and even possibly as a list, something like `references`, `public_sources`, of maybe `public_references`. Maybe even use a constant like `REFERENCES` to denote that the value is not expected to ever be manipulated in by an instance.

Reply:
I like the idea of refactoring the reference `uri` to `references`/`REFERENCES`

Reply:
Went with `doc_uri`  for now, happy to revisit down the line.